src/twisted/scripts/trial.py

```diff
@@ -36,6 +36,39 @@
 }
 
 
+def _autoJobs() -> int:
+    """
+    Heuristically guess the number of job workers to run.
+
+    When ``os.process_cpu_count()`` is available (Python 3.13+),
+    return the number of logical CPUs usable by the current
+    process. This respects the ``PYTHON_CPU_COUNT`` environment
+    variable and/or ``python -X cpu_count`` flag.
+
+    Otherwise, if ``os.sched_getaffinity()`` is available (on some
+    Unixes) this returns the number of CPUs this process is
+    restricted to, under the assumption that this affinity will
+    be inherited.
+
+    Otherwise, consult ``os.cpu_count()`` to get the number of
+    logical CPUs.
+
+    Failing all else, return 1.
+
+    @returns: A strictly positive integer.
+    """
+    number: Optional[int]
+    if getattr(os, "process_cpu_count", None) is not None:
+        number = os.process_cpu_count()  # type: ignore[attr-defined]
```

Comment:
Maybe add a comment on why process_cpu_count is conditional


```suggestion
    if getattr(os, "process_cpu_count", None) is not None:
        # On Python 3.13 or newer.
        number = os.process_cpu_count()  # type: ignore[attr-defined]
```

Reply:
This is covered in the docstring.